pipeline/matching/core/analytics.py

```diff
@@ -0,0 +1,27 @@
+from pipeline.matching.core import MatchingEngine
+from pipeline.types import MatchingInput, MatchingInternalAnalytics
+
+
+class CollectInternalAnalytics:
+    def __init__(self, name, matching_engine):
+        self.name: str = name
+        self.matching_engine: MatchingEngine = matching_engine
+
+    def n_proposed_matches_per_user(
+        self, inp: MatchingInput
+    ) -> MatchingInternalAnalytics:
```

Comment:
Does this need to be a class with a function or can it just be a function? What do you think are the pros and cons?

Reply:
Well...I was thinking that if we are going to include more metrics in the future, then they would fall under this CollectInternalAnalytics class. While it can be its own function, if we do plan to add more metrics it will help in keeping things more organized. Not to mention using a class is a good way to represent a collection of attributes and methods that may be used repeatedly in other places.

Reply:
I agree with you, let's keep it as a class then!

One thing to think about is merge conflicts, sometimes if multiple people are writing new metrics in the same class, they might create conflicts for each other that slow down changes and reviews. But, we can learn how to handle those.